syntax = "proto3";

package transaction_verification;

service TransactionVerificationService {
    rpc VerifyTransaction (TransactionRequest) returns (TransactionResponse);
}

message TransactionRequest {
  User user = 1;
  CreditCard credit_card = 2;
  string user_comment = 3;
  repeated Item items = 4;
  Address billing_address = 5;
  string shipping_method = 6;
  bool gift_wrapping = 7;
  bool terms_accepted = 8;
}

message User {
  string name = 1;
  string contact = 2;
}

message Item {
  string name = 1;
  int32 quantity = 2;
}

message CreditCard {
  string number = 1;
  string expiration_date = 2;
  string cvv = 3;
}

message Address {
  string street = 1;
  string city = 2;
  string state = 3;
  string zip = 4;
  string country = 5;
}

message TransactionResponse {
  bool is_valid = 1;
  repeated string errors = 2;
}
